File: aulas/pii2425/code/primes.py
LIMIT = 10000000

def is_prime(n):
    if n < 2: return False
    if n % 2 == 0: return n == 2
    if n % 3 == 0: return n == 3
    p = 5
    while p * p <= n:
        if n % p == 0: return False
        p += 2
    return True

def main():
    
    count = 0
    for n in range(1, LIMIT):
        if is_prime(n):
            count += 1
    print(f"There are {count} primes less than {LIMIT}")

if __name__ == "__main__":
    main()

File: aulas/pii2425/practical/primes.c
#include <stdio.h>
#include <stdbool.h>

#define LIMIT 10000000

bool is_prime(int n) {
    if (n < 2) return false;
    if (n % 2 == 0) return n == 2;
    if (n % 3 == 0) return n == 3;
    int p = 5;
    while (p * p <= n) {
      if (n % p == 0) return false;
      p += 2;
    }
    return true;
}

int main() {

  int count = 0;
  for (int n = 1; n < LIMIT; n++)
    if (is_prime(n))
      count++;
  printf("There are %d primes less than %d\n", count, LIMIT);
  
  return 0;
}

